cpu/cpu_windows.go

```diff
@@ -200,30 +204,83 @@ type systemInfo struct {
 	wProcessorRevision          uint16
 }
 
+type groupAffinity struct {
+	mask     uintptr // https://learn.microsoft.com/it-it/windows-hardware/drivers/kernel/interrupt-affinity-and-priority#about-kaffinity
+	group    uint16
+	reserved [3]uint16
+}
+
+// https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-processor_relationship
+type processorRelationship struct {
+	flags          byte
+	efficientClass byte
+	reserved       [20]byte
+	groupCount     uint16
+	groupMask      [1]groupAffinity
+}
+
+// https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-system_logical_processor_information_ex
+type systemLogicalProcessorInformationEx struct {
+	Relationship uint32
+	Size         uint32
+	Processor    processorRelationship
+}
+
+func getPhysicalCoreCount(ctx context.Context) (int, error) {
+	var length uint32
+	const relationAll = 0xffff
+	const relationProcessorCore = 0x0
+
+	// First call to determine the required buffer size
+	_, _, err := getLogicalProcessorInformationEx.Call(uintptr(relationAll), 0, uintptr(unsafe.Pointer(&length)))
+	if err != nil && !errors.Is(err, windows.ERROR_INSUFFICIENT_BUFFER) {
+		return 0, fmt.Errorf("failed to get buffer size: %w", err)
+	}
+
+	// Allocate the buffer
+	buffer := make([]byte, length)
+
+	// Second call to retrieve the processor information
+	_, _, err = getLogicalProcessorInformationEx.Call(uintptr(relationAll), uintptr(unsafe.Pointer(&buffer[0])), uintptr(unsafe.Pointer(&length)))
+	if err != nil && !errors.Is(err, windows.NTE_OP_OK) {
+		return 0, fmt.Errorf("failed to get logical processor information: %w", err)
+	}
+
+	// Iterate through the buffer to count physical cores
+	offset := uintptr(0)
+	ncpus := 0
+	for offset < uintptr(length) {
+		select {
+		case <-ctx.Done():
```

Comment:
Did you extract [this part in the `common.WMIQueryWithContext()` function](https://github.com/shirou/gopsutil/blob/ef5056a446730347150887a7e485da69bd97ae3e/internal/common/common_windows.go#L213-L214)? With the function being way faster without wmi, is this still really needed?

Reply:
Implementing cancellation individually within each platform-specific function, rather than in a shared/common function, would be difficult and inconsistent. So while I appreciate your effort, please remove this cancellation logic.

This also relates to [this comment](https://github.com/shirou/gopsutil/pull/1837#issuecomment-2816997435), but it would be ideal if we could implement some kind of common function for handling cancellation.

Reply:
@Lomanic Could you check the update ?